pkg/migrations/sql/idempotent/base.sql

```diff
@@ -204,27 +212,35 @@ DECLARE
   label_id INT;
   compressed_hypertable_name text;
 BEGIN
-   EXECUTE format('CREATE TABLE SCHEMA_DATA.%I(time TIMESTAMPTZ NOT NULL, value DOUBLE PRECISION NOT NULL, series_id BIGINT NOT NULL) WITH (autovacuum_vacuum_threshold = 50000, autovacuum_analyze_threshold = 50000)',
-                    NEW.table_name);
-   EXECUTE format('GRANT SELECT ON TABLE SCHEMA_DATA.%I TO prom_reader', NEW.table_name);
-   EXECUTE format('GRANT SELECT, INSERT ON TABLE SCHEMA_DATA.%I TO prom_writer', NEW.table_name);
-   EXECUTE format('GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE SCHEMA_DATA.%I TO prom_modifier', NEW.table_name);
-   EXECUTE format('CREATE UNIQUE INDEX data_series_id_time_%s ON SCHEMA_DATA.%I (series_id, time) INCLUDE (value)',
-                    NEW.id, NEW.table_name);
+
+   -- Note: if the inserted metric is a view, handle the necessary permissions and exit.
+   IF NEW.is_view THEN
```

Comment:
Isn't it misleading in terms of naming of this function? If our task is to just give permissions to te views, wouldn't it be better to have another function as `assign_permissions_to_metric_views`

Reply:
I don't see a reason to change since it mostly ignores doing anything in case of a view (except for some permission handling).